src/omit.ts

```diff
@@ -28,15 +28,15 @@ export function omit<T extends object, K extends keyof T>(
  * @dataLast
  * @category Object
  */
-export function omit<K extends PropertyKey>(
+export function omit<T extends Record<PropertyKey, unknown>, K extends keyof T>(
   propNames: ReadonlyArray<K>
-): <T extends object>(data: T) => Omit<T, K>;
+): (data: T) => Omit<T, K>;
 
 export function omit() {
   return purry(_omit, arguments);
 }
 
-function _omit<T extends object, K extends keyof T>(
+function _omit<T extends Record<PropertyKey, unknown>, K extends keyof T>(
```

Comment:
Similarly
```suggestion
function _omit<T, K extends keyof T>(
```

Reply:
I need at least an `object` type or the implementation code will yell at me:

![image](https://github.com/remeda/remeda/assets/1144075/a367cd2a-c49b-4a15-aeff-a77a69342a10)